webapp/src/pages/JoinChatroomPage/JoinChatroomPage.tsx

```diff
@@ -0,0 +1,131 @@
+import { useCallback, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import {
+  Button,
+  ButtonVariant,
+  Header,
+  IconVariant,
+  Input,
+  InputVariant,
+  LargeAvatar,
+  Loading,
+  Text,
+  TextVariant,
+} from '../../shared/components';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+import { ResponseError } from '../../shared/generated';
+import { useData } from '../../shared/hooks/UseData';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { chatApi } from '../../shared/services/ApiService';
+import {
+  CHATROOM_EP_URL,
+  CHATROOM_URL,
+  WILDCARD_AVATAR_URL,
+} from '../../shared/urls';
+import NotFoundPage from '../NotFoundPage/NotFoundPage';
+import './JoinChatroomPage.css';
+
+export default function JoinChatroomPage() {
+  const { chatroomId } = useParams();
+
+  if (!chatroomId) {
+    return null;
+  }
+  return <JoinChatroom chatroomId={chatroomId} />;
+}
+
+type JoinChatroomProps = {
+  chatroomId: string;
+};
+
+function JoinChatroom({ chatroomId }: JoinChatroomProps) {
+  const { goBack } = useNavigation();
+  const getChatroom = useCallback(
+    () => chatApi.chatControllerGetChatroomById({ id: chatroomId }),
+    [chatroomId],
+  );
+  const { data: chatroom, isLoading: isChatroomLoading } = useData(getChatroom);
+  const isProtectedChatroom = chatroom?._public === false;
+  const [password, setPassword] = useState('');
+  const { warn, notify } = useNotificationContext();
+  const { navigate } = useNavigation();
+
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (
+    event,
+  ) => {
+    event.preventDefault();
+    try {
+      await chatApi.chatControllerCreateChatroomMember({
+        chatroomId: chatroomId,
+        joinChatroomDto: {
+          password: isProtectedChatroom ? password : undefined,
+        },
+      });
+      notify('You joined a chatroom');
+      navigate(`${CHATROOM_URL}/${chatroomId}`, { replace: true });
```

Comment:
What does `{ replace: true }` do?

Reply:
A call to `navigate` pushes a new entry into the history stack so the user can click the back button to get back to the page. If we pass `{ replace: true }`, we replace the current entry with the new one.
For example, if the user clicks the "join" button, we replace the join chatroom screen with the chatroom screen. Then when they click the back button, they don't see the join page again.